src/controllers/movies.controller.ts

```diff
@@ -0,0 +1,96 @@
+import { Request } from "express";
+
+import { IMovieDocument } from "../models";
+import { MoviesService } from "../services/movies.service";
+
+export class MoviesController {
+  constructor(private moviesService: MoviesService) {}
+
+  async getMovie(req: Request): Promise<IMovieDocument | null> {
+    const { id } = req.params;
+    const movie = await this.moviesService.findById({ id });
+    return movie;
+  }
+
+  async createMovie(req: Request): Promise<IMovieDocument> {
+    const newMovie = await this.moviesService.createMovie({
+      data: req.body,
+    });
+    return newMovie;
+  }
+
+  async updateMovie(req: Request): Promise<IMovieDocument | null> {
+    const { id } = req.params;
+    const movie = await this.moviesService.findByIdAndUpdate({
+      id,
+      data: req.body,
+    });
+    return movie;
+  }
+
+  async updateFullMovie(req: Request): Promise<IMovieDocument | null> {
+    const { id } = req.params;
+    const movie = await this.moviesService.findByIdAndUpdateFull({
+      id,
+      data: req.body,
+    });
+    return movie;
+  }
+
+  async deleteMovie(req: Request): Promise<IMovieDocument | null> {
+    const { id } = req.params;
+    const movie = await this.moviesService.deleteMovie({ id });
```

Comment:
For improvements, you can add checking if "movie" is present.
Because you can delete several times the same movie and receive 200 status

Reply:
Thanks, added movie/genre check for Delete request